### PR TITLE
Add a CONTRIBUTING.md and STYLEGUIDE.md

### PR DESCRIPTION
**Pull request by garakmon**
Feedback greatly appreciated.

Also I've been sort of in a rush today so I still have to read it over later, I might have made mistakes.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+
+Search for the file in the linker script `ld_script.txt`:
+
+```
+    asm/fire.o(.text);
+    asm/electric.o(.text);
+    asm/ice.o(.text);
+```
+
+and add the new C file BEFORE the assmebly file, since we are starting to
```
Comment (ultima-soul) [inline CONTRIBUTING.md]: assmebly -> assembly

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+We begin by trying to identify the function signature (the return type, and
+any arguments).
+
+To determine the return type of the function, we look at the bottom where the 
+function ends. We see `bx r0` which quickly tells us that our function is 
+`void`.
+
+Other branches you might see are `bx r1` or `bx lr`. Seeing these means you'll 
+need to infer the return type from the way it is called, and the values in the
+registers at the end, but it will never.
```
Comment (ultima-soul) [inline CONTRIBUTING.md]: Sentence seems incomplete here.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+tells us that this argument is probably a struct pointer. More context confirms
+that is it a pointer to a Sprite.
+
+Our C file now looks like this:
+
+```C
+#include "global.h"
+#include "sprite.h"
+
+void sub_80ADBEC(struct Sprite *sprite)
```
Comment (ultima-soul) [inline CONTRIBUTING.md]: Italicization starts being used here for some reason and ends at line 244.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+    sprite->callback = sub_80ADC3C;
+```
+
+Putting this all together, we get
+
+```C
+#include "global.h"
+#include "sprite.h"
+
+extern void sub_80ADC3C(struct Sprite *);
```
Comment (ultima-soul) [inline CONTRIBUTING.md]: Italicization starts here and ends at line 411.

**Comment by PikalaxALT** [comment]
Perhaps these should be placed in a github.io page rather than separately in each repository.

**Comment by garakmon** [comment]
Uh what about the wiki? I feel like pokeemerald and pokeruby are far enough along to not have a use for this.

**Comment by PikalaxALT** [comment]
I'd prefer to have these documents in a centralized location for the sake of current and future projects.

**Comment by garakmon** [comment]
I don't disagree, I just don't know where that would be.

**Comment by PikalaxALT** [comment]
As alluded to, we could create a `pret.github.io` page to organize shared documents and tutorials.

File: CONTRIBUTING.md (hunk 3 above)
Comment (PikalaxALT) [reply CONTRIBUTING.md]: That's just GitHub being a dum-dum

**Comment by coltongit** [comment]
Considering an issue in Emerald was opened for this as well, could that be looked at too? https://github.com/pret/pokeemerald/issues/598

**Pull request closed without merging**